### PR TITLE
Rewrite FocusableIframe as hook API

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

This component can be rewritten as a behavioural hook and used in both the Sandbox and WpEmbedPreview components.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/compose/src/hooks/use-focusable-iframe/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+## Usage
+
+Use with an `iframe`. You may pass `onFocus` directly as the callback to be invoked when the iframe receives focus, or on an ancestor component since the event will bubble.
+
+```jsx
+import { useRef } from '@wordpress/element';
+import { useFocusableIframe } from '@wordpress/compose';
+
+const MyFocusableIframe = () => {
+    const ref = useRef( ref );
```
Comment (kevin940726) [inline packages/compose/src/hooks/use-focusable-iframe/README.md:14]: I believe it should be...?

```suggestion
    const ref = useRef();
```

File: packages/compose/src/hooks/use-focusable-iframe/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    useFocusableIframe( ref );
+
+    return(
+        <iframe
+            src="/my-iframe-url"
+            onFocus={ () => console.log( 'iframe is focused' ) }
```
Comment (kevin940726) [inline packages/compose/src/hooks/use-focusable-iframe/README.md:21]: I believe we should also add `ref` here?

```suggestion
        <iframe
        	ref={ ref }
            src="/my-iframe-url"
            onFocus={ () => console.log( 'iframe is focused' ) }
```

File: packages/compose/src/hooks/use-focusable-iframe/index.js [hunk 3]
```diff
@@ -3,25 +3,28 @@
...
 
 		/**
 		 * Checks whether the iframe is the activeElement, inferring that it has
-		 * then received focus, and calls the `onFocus` prop callback.
+		 * then received focus, and dispatches a focus event.
 		 */
 		function checkFocus() {
-			if ( ownerDocument.activeElement !== iframe ) {
-				return;
+			if ( ownerDocument.activeElement !== ref.current ) {
```
Comment (kevin940726) [inline packages/compose/src/hooks/use-focusable-iframe/index.js:23]: Code here doesn't match the comments above. It should be `===` here?

```suggestion
			if ( ownerDocument.activeElement === ref.current ) {
```

**Review by kevin940726 — CHANGES_REQUESTED** [review]

File: packages/compose/src/hooks/use-focusable-iframe/README.md (hunk 2 above)
Comment (ellatrix) [reply packages/compose/src/hooks/use-focusable-iframe/README.md:21]:
```suggestion
        <iframe
            ref={ ref }
            src="/my-iframe-url"
            onFocus={ () => console.log( 'iframe is focused' ) }
```

**Review by kevin940726 — APPROVED** [review]

**Pull request merged**